hydra/grammar/OverrideParser.g4

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+// Regenerate parser by running 'python setup.py antlr' at project root.
+// If you make changes here be sure to update the documentation (and update the grammar in command_line_syntax.md)
```

Comment:
Updated in 8023c495

> Speaking of which, did document anything?

Oh, that's a very good point, I assumed no update of the doc was needed since it was a bug fix, but I forgot the grammar was copied into the doc. I added the doc update to 8023c495

I wanted to check with you before documenting the whitespace behavior though: right now there is nothing in the doc about unquoted strings. It is written in such a way that it seems like if it's not quoted, then it needs to be a single primitive (as if the grammar didn't have this `+` in the primitive rule). Do you want me to add a subsection on unquoted strings?